### DISCUSSION TITLE
Does it work with raspberry pi pico?

### DISCUSSION THREAD

**Discussion by TomP4211** [arkhipenko/TaskScheduler#167] [Q&A] [answered]
I know it says it works with Raspberry Pi. But does it work also with the raspberry pi pico?

**Comment by arkhipenko** [comment]
If you provide your own millis() method in an Arduino.h file it totally should. 
This is probably the only Arduino dependency of the entire library.

**Reply by TomP4211** [reply]
Thank you for the fast reply! Would something like this work?: 

static uint64_t startTime = 0;

void initMillis() {
  startTime = time_us_64();
}

uint64_t millis() {
  uint64_t currentTime = time_us_64();
  return (currentTime - startTime) / 1000;
}

Basically converting the time from microseconds (time_us_64) to miliseconds? Is there anything else i need to look out for?

**Reply by arkhipenko** [reply]
millis() should be uint32_t, other than that, that's it.

**Reply by arkhipenko** [reply, marked as answer]
You might need to also provide micros(), but it could be empty to just satisfy the dependency if you are not using it. 

Or just convert (time_us_64) to uint32_t and return that.

**Reply by TomP4211** [reply]
Thank you so much! I will try to do that.
